src/solvers/prop/prop.h

```diff
@@ -94,7 +94,8 @@ class propt:public messaget, public prop_assignmentt
   // solving
   virtual const std::string solver_text()=0;
   enum class resultt { P_SATISFIABLE, P_UNSATISFIABLE, P_ERROR };
-  virtual resultt prop_solve()=0;
+  resultt operator()() { return prop_solve(); }
```

Comment:
Please add documentation: `operator()` shortens the code, but it hardly seems obvious what an `operator()` of a class named `propt` is supposed to do. `prop_solve` at least made clear that some solver is being invoked.

Reply:
Since it looks like we don't use `propt` as functors anywhere I don't see the need for defining a `()` operator. 
Indeed I would prefer a more explicit name like `solve()`.